src/models/employees/employee_preferences/employees_shifts_preferences.py

```diff
@@ -1,14 +1,17 @@
 from dataclasses import field
+from typing import Union
 
 import pydantic
 
-from src.models.employees.employee_preferences.no_preferences import NoPreference
-from src.models.employees.employee_preferences.shifts_preference import ShiftsPreference
+from .combine_preference import CombinePreference
+from .date_time_range_preference_ import DateTimeRangePreference
+from .no_preferences import NoPreference
 from src.models.solution.pydantic_config import ConfigPydanticDataclass
+from .shifts_preference_by_id import ShiftIdPreference
 
 
 @pydantic.dataclasses.dataclass(config=ConfigPydanticDataclass)
 class EmployeesShiftsPreferences:
-    shifts_cannot_work: ShiftsPreference = field(default_factory=NoPreference)
-    shifts_prefer_not_to_work: ShiftsPreference = field(default_factory=NoPreference)
-    shifts_wants_to_work: ShiftsPreference = field(default_factory=NoPreference)
+    shifts_cannot_work: Union[CombinePreference, DateTimeRangePreference, NoPreference, ShiftIdPreference] = field(default_factory=NoPreference)
```

Comment:
extract the type definition `Union[CombinePreference, DateTimeRangePreference, NoPreference, ShiftIdPreference]` to a variable